k2/csrc/ragged.h

```diff
@@ -465,6 +465,20 @@ struct Ragged {
     return Ragged<T>(new_shape, new_values);
   }
 
+#define ToType(type, name)                                  \
```

Comment:
Hm, I was thinking you'd just overload the to() function to accept a torch type?
That's how I normally do type conversions in torch, e.g. tensor.to(torch.float32) or something.
This code is kind of hard to read and hard to discover.

Reply:
> That's how I normally do type conversions in torch, e.g. tensor.to(torch.float32) or something.

Yes, the current implementation supports that. Valid usages are:
```python
# create a ragged_tensor
ragged_tensor = ragged_tensor.to(torch.device('cuda', 0))
ragged_tensor = ragged_tensor.to(torch.int32)
```

> Hm, I was thinking you'd just overload the to() function to accept a torch type?

`torch.dtype` and `torch.device` cannot be accessed in C++, I think. Both of them can be accessed in
C++ using `py::object` and that's why I overload `k2::To`: (1) One function is to handle `torch.device`
(2) The other function is to handle `torch.dtype`.

The reason I am using `ToType` is to implement something like converting `Ragged<float>` to `Ragged<int32_t>`.
I tried to replace the macro with a template, but it throws some errors mentioned in the comment that I've no
idea how to solve them.